FreeRTOS_TCP_IP.c

```diff
@@ -1336,19 +1346,27 @@
                             break;
                         }
 
-                        uxResult = prvSingleStepTCPHeaderOptions( pucPtr, uxOptionsLength, pxSocket, xHasSYNFlag );
+                        lResult = prvSingleStepTCPHeaderOptions( pucPtr, uxOptionsLength, pxSocket, xHasSYNFlag );
 
-                        if( uxResult == 0UL )
+                        if( lResult < 0 )
                         {
+                            xReturn = pdFAIL;
                             break;
                         }
 
-                        uxOptionsLength -= uxResult;
-                        pucPtr = &( pucPtr[ uxResult ] );
+                        if( lResult == 0UL )
```

Comment:
Sorry to be boring:
~~~c
-    if( lResult == 0UL )
+    if( lResult == 0 )
~~~
The `L` suffix may lead to a 64-bit value.

Reply:
Done! Thanks!